### PR TITLE
Refactor KNNVectorFieldType from KNNVectorFieldMapper to a separate class for better readability.

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Refactor KNNVectorFieldType from KNNVectorFieldMapper to a separate class for better readability. This change doesn't change any functionality, it's just Mapper class was becoming too big and refactoring this class out helps reducing the size of the file and things to look at once.

### Related Issues
NA

### Check List
- [X] Commits are signed per the DCO using `--signoff`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by heemin32 — DISMISSED** [review]

**Comment by jmazanec15** [comment]
@navneet1v did you update the required tests?

**Comment by navneet1v** [comment]
@jmazanec15 I updated. Still seems some error, let me check

File: src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.opensearch.knn.common.KNNConstants.DEFAULT_VECTOR_DATA_TYPE_FIELD;
+import static org.opensearch.knn.index.mapper.KNNVectorFieldMapperUtil.deserializeStoredVector;
+
+/**
+ * An Opensearch field Type which to create the Lucene field.
```
Comment (jmazanec15) [inline src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java:34]: nit: MappedFieldType doesnt create the Lucene fieldType. It seems like its used more on search side of things.

**Comment by navneet1v** [comment]
Few classes were missed. Fixing them

File: src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java (hunk 1 above)
Comment (navneet1v) [reply src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java]: updated

**Review by heemin32 — APPROVED** [review]

**Comment by navneet1v** [comment]
BWC tests are failing because of the version upgrade. @shatejas when we are fixing the BWCs for the 2.17 version?

**Review by jmazanec15 — APPROVED** [review]

**Pull request merged**